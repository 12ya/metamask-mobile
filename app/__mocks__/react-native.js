import { Linking, DeviceEventEmitter } from 'react-native';

Linking.addEventListener = jest.fn();
Linking.removeEventListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeEventListener
  if (typeof callback === 'function') {
    callback();
  }
});
Linking.openURL = jest.fn();
Linking.canOpenURL = jest.fn();
Linking.getInitialURL = jest.fn();

DeviceEventEmitter.addListener = jest.fn();
DeviceEventEmitter.removeListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeListener
  if (typeof callback === 'function') {
    callback();
  }
});

export default { Linking, DeviceEventEmitter };
