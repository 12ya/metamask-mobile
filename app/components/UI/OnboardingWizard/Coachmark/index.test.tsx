import React from 'react';
import { render } from '@testing-library/react-native';
import Coachmark from './';

describe('Coachmark', () => {
  it('should render correctly', () => {
    const { toJSON } = render(
      <Coachmark
        content={'content'}
        title={'title'}
        currentStep={1}
        topIndicatorPosition={'topLeft'}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
