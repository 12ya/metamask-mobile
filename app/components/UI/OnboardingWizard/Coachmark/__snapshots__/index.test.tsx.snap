// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coachmark should render correctly 1`] = `
<ForwardRef(AnimatedComponentWrapper)
  style={
    [
      {},
      {
        "opacity": 0,
      },
    ]
  }
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "bottom": -2,
        "marginBottom": 10,
        "marginLeft": 30,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
          "borderBottomColor": "#0376C9",
          "borderBottomWidth": 12,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 15,
          "borderRightColor": "transparent",
          "borderRightWidth": 15,
          "borderStyle": "solid",
          "height": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#0376C9",
          "borderRadius": 8,
          "padding": 18,
        },
        {},
      ]
    }
  >
    <View>
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "#FCFCFC",
            "fontFamily": "EuclidCircularB-Bold",
            "fontSize": 18,
            "fontWeight": "600",
          }
        }
      >
        title
      </Text>
    </View>
    content
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <StyledButton
        containerStyle={
          [
            {
              "height": 45,
              "padding": 5,
              "width": 75,
            },
            {
              "left": 0,
            },
          ]
        }
        disabledContainerStyle={
          {
            "opacity": 0.6,
          }
        }
        onPress={[Function]}
        styleDisabled={
          {
            "opacity": 0.6,
          }
        }
        type="inverse-transparent"
      >
        Back
      </StyledButton>
      <View
        style={
          {
            "alignSelf": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          key="1"
          style={
            [
              {
                "backgroundColor": "#FCFCFC",
                "borderRadius": 3,
                "height": 6,
                "margin": 3,
                "opacity": 0.4,
                "width": 6,
              },
              {
                "opacity": 1,
              },
            ]
          }
        />
        <View
          key="2"
          style={
            [
              {
                "backgroundColor": "#FCFCFC",
                "borderRadius": 3,
                "height": 6,
                "margin": 3,
                "opacity": 0.4,
                "width": 6,
              },
              {},
            ]
          }
        />
        <View
          key="3"
          style={
            [
              {
                "backgroundColor": "#FCFCFC",
                "borderRadius": 3,
                "height": 6,
                "margin": 3,
                "opacity": 0.4,
                "width": 6,
              },
              {},
            ]
          }
        />
        <View
          key="4"
          style={
            [
              {
                "backgroundColor": "#FCFCFC",
                "borderRadius": 3,
                "height": 6,
                "margin": 3,
                "opacity": 0.4,
                "width": 6,
              },
              {},
            ]
          }
        />
        <View
          key="5"
          style={
            [
              {
                "backgroundColor": "#FCFCFC",
                "borderRadius": 3,
                "height": 6,
                "margin": 3,
                "opacity": 0.4,
                "width": 6,
              },
              {},
            ]
          }
        />
      </View>
      <StyledButton
        containerStyle={
          [
            {
              "height": 45,
              "padding": 5,
              "width": 75,
            },
            {
              "right": 0,
            },
          ]
        }
        disabledContainerStyle={
          {
            "opacity": 0.6,
          }
        }
        onPress={[Function]}
        styleDisabled={
          {
            "opacity": 0.6,
          }
        }
        type="inverse"
      >
        Got it!
      </StyledButton>
    </View>
  </View>
</ForwardRef(AnimatedComponentWrapper)>
`;
